src/hooks/core/useCoreChatSystemHook.ts

```diff
@@ -219,24 +239,41 @@ const useCoreChatSystemHook = () => {
       websocketMessage?.user?.id === anonProfileId;
 
     if (!isMyMessage) {
-      const chat = ChatSchema.fromWebsocketObject(websocketData);
-      await chat.save(localDb);
-    }
-
-    try {
-      websocketData?.originalMembers?.forEach(async (member) => {
-        const userMember = UserSchema.fromMemberWebsocketObject(member, websocketData?.channel?.id);
-        await userMember.saveOrUpdateIfExists(localDb);
+      queue.addPriorityJob({
+        operationLabel: DatabaseOperationLabel.CoreChatSystem_SaveChat,
+        id: `${websocketData?.channel?.id}-${websocketData?.message?.id}`,
+        task: async () => {
+          const chat = ChatSchema.fromWebsocketObject(websocketData);
+          await chat.save(localDb);
+        },
+        priority: QueueJobPriority.HIGH
       });
-    } catch (e) {
-      console.log('error on memberSchema');
-      console.log(e);
     }
 
-    refresh('channelList');
-    refresh('chat');
-    refresh('channelInfo');
-    refresh('channelMember');
+    queue.addPriorityJob({
+      operationLabel: DatabaseOperationLabel.CoreChatSystem_SaveUserMember,
+      id: `${websocketData?.channel?.id}`,
+      priority: QueueJobPriority.HIGH,
+      task: async () => {
+        try {
+          websocketData?.originalMembers?.forEach(async (member) => {
+            const userMember = UserSchema.fromMemberWebsocketObject(
+              member,
+              websocketData?.channel?.id
+            );
+            await userMember.saveOrUpdateIfExists(localDb);
+          });
+        } catch (e) {
+          console.log('error on memberSchema');
+          console.log(e);
+        }
+
+        refresh('channelList');
+        refreshWithId('chat', websocketData?.channel?.id);
+        refresh('channelInfo');
+        refresh('channelMember');
+      }
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes here involve adding database operations to a priority queue and refreshing specific data by id. These changes can improve efficiency but should be thoroughly tested to ensure they don't introduce unexpected behavior.

```diff
- const chat = ChatSchema.fromWebsocketObject(websocketData);
- await chat.save(localDb);
+ queue.addPriorityJob({
+   operationLabel: DatabaseOperationLabel.CoreChatSystem_SaveChat,
+   id: `${websocketData?.channel?.id}-${websocketData?.message?.id}`,
+   task: async () => {
+     const chat = ChatSchema.fromWebsocketObject(websocketData);
+     await chat.save(localDb);
+   },
+   priority: QueueJobPriority.HIGH
+ });

- websocketData?.originalMembers?.forEach(async (member) => {
-   const userMember = UserSchema.fromMemberWebsocketObject(member, websocketData?.channel?.id);
-   await userMember.saveOrUpdateIfExists(localDb);
- });
+ queue.addPriorityJob({
+   operationLabel: DatabaseOperationLabel.CoreChatSystem_SaveUserMember,
+   id: `${websocketData?.channel?.id}`,
+   priority: QueueJobPriority.HIGH,
+   task: async () => {
+     try {
+       websocketData?.originalMembers?.forEach(async (member) => {
+         const userMember = UserSchema.fromMemberWebsocketObject(
+           member,
+           websocketData?.channel?.id
+         );
+         await userMember.saveOrUpdateIfExists(localDb);
+       });
+     } catch (e) {
+       console.log('error on memberSchema');
+       console.log(e);
+     }
+
+     refresh('channelList');
+     refreshWithId('chat', websocketData?.channel?.id);
+     refresh('channelInfo');
+     refresh('channelMember');
+   }
+ });
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->